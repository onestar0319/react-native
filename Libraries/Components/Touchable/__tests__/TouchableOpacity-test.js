/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @format
 * @emails oncall+react_native
 */

'use strict';

const React = require('react');
const Text = require('../../../Text/Text');
const TouchableOpacity = require('../TouchableOpacity');

const render = require('../../../../jest/renderer');

describe('TouchableOpacity', () => {
  it('renders correctly', () => {
    const instance = render.create(
      <TouchableOpacity style={{}}>
        <Text>Touchable</Text>
      </TouchableOpacity>,
    );

    expect(instance.toJSON()).toMatchSnapshot();
  });

  it('has displayName', () => {
    expect(TouchableOpacity.displayName).toEqual('TouchableOpacity');
  });
});
