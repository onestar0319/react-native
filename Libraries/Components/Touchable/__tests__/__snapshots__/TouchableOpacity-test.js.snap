// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TouchableOpacity renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Text>
    Touchable
  </Text>
</View>
`;
